### DISCUSSION TITLE
The pixel formats of freerdp and cairo are not aligned？

### DISCUSSION THREAD

**Discussion by RyanCode** [FreeRDP/FreeRDP#9039] [Q&A] [answered]
I am using the cairo library to analyze the bitmap of freerdp. The pixel format specified when initializing gdi is PIXEL_FORMAT_ARGB32. When using cairo to analyze, I use CAIRO_FORMAT_ARGB32. At this time, the image is distorted and the analysis fails. When I implement the PIXEL_FORMAT_BGRA32 format, the graphic parsing is normal, why? Looking forward to your reply.

**Comment by akallabeth** [comment, marked as answer]
basically naming.
depending on if you see that big or little endian `ARGB (little) == BGRA (big)`
